ReplyAdaptabilityCodelab/app/src/main/java/com/example/reply/ui/WindowStateUtils.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.reply.ui
+
+import android.app.Activity
+import android.graphics.Rect
+import androidx.annotation.VisibleForTesting
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.toComposeRect
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import androidx.window.layout.FoldingFeature
+import androidx.window.layout.WindowMetricsCalculator
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+
+/**
+ * Opinionated set of viewport breakpoints
+ *     - Compact: Most phones in portrait mode
+ *     - Medium: Most foldables and tablets in portrait mode
+ *     - Expanded: Most tablets in landscape mode
+ *
+ * More info: https://m3.material.io/foundations/adaptive-design/overview
+ */
+enum class WindowSize { COMPACT, MEDIUM, EXPANDED }
```

Comment:
We need a TODO here to use `androidx.compose.material3:material3-window-size-class`:
https://developer.android.com/jetpack/androidx/releases/compose-material3#1.0.0-alpha10
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material3/material3-window-size-class/

Reply:
Yes I plan to add it soon.

Reply:
Let's directly use the m3 window-size-classes. I prefer not to merge the codelab without this change.